src/dialects/mysql/schema/columncompiler.js

```diff
@@ -113,7 +118,18 @@ assign(ColumnCompiler_MySQL.prototype, {
   collate(collation) {
     return collation && `collate '${collation}'`
   }
-
+  
 })
 
+const MIN_JSON_VERSION = semver.SemVer('5.7.8');
+function jsonColumn(client) {
+  if(!client.version) {
+    return 'json'; // this is here jsut for tests
+  } else if(semver.valid(client.version) && semver.gte(client.version, MIN_JSON_VERSION)) {
```

Comment:
I suppose this same check would work also for mariadb removing need for changes in its column compiler...